samza-operator/src/test/java/org/apache/samza/example/BroadcastGraph.java

```diff
@@ -57,26 +71,36 @@ public long getTimestamp() {
   }
 
   @Override
-  public void transform(Map<SystemStreamPartition, MessageStream<IncomingSystemMessageEnvelope>> messageStreams) {
-    BiFunction<JsonMessageEnvelope, Integer, Integer> sumAggregator = (m, c) -> c + 1;
-    messageStreams.values().forEach(entry -> {
-        MessageStream<JsonMessageEnvelope> inputStream = entry.map(this::getInputMessage);
+  public StreamGraph create(Config config) {
+    StreamGraphImpl graph = new StreamGraphImpl();
 
-        inputStream.filter(this::myFilter1)
-          .window(Windows.tumblingWindow(Duration.ofMillis(100), sumAggregator)
+    BiFunction<JsonMessageEnvelope, Integer, Integer> sumAggregator = (m, c) -> c + 1;
+    inputs.forEach(entry -> {
+        MessageStream<JsonMessageEnvelope> inputStream = graph.<Object, Object, InputMessageEnvelope>createInStream(new StreamSpec() {
```

Comment:
Not sure if it's easier this way, but it doesn't seem to me that we need a separate StreamSpec instance for every SSP (emphasis on partition)